### ISSUE TITLE
use accessibility roles as appropriate

### ISSUE THREAD

**Issue by lsloan** [tl-its-umich-edu/my-learning-analytics#1165]
## Problem

As written in the original issue (#807):

> Regarding the main container, the accessibility review suggests adding the `role="main"` attribute-value pair to the HTML element serving as the main container.

When MyLA is used as an LTI and embedded in a LMS page, the LMS applies the role of `main` that applies to the MyLA content.  (At least, Canvas does, it hasn't been tested with other LMSes.)  That handles most of the accessibility role issues.  If MyLA will always be used this way, then there will be no problem.

However, if MyLA is used as a standalone application, roles are not being set.  In that case, the header needs a `header` or `navigation` role, the content needs a `main` role, and the footer needs a `footer` role.  To implement that, a parameter would need to be passed from the application to the presentation layer to indicate whether it is operating in standalone mode and should set these roles.

It is unclear whether adding these roles *_always_* (that is, while it is embedded *_and_* while it is standalone) would adversely affect the embedded mode.  While embedded, the MyLA content is contained in an `iframe` element.  `iframe` content essentially has its own DOM, etc., so adding a `main` role to elements within it probably would not violate the existence of the `main` role in the parent page.  However, each brand of web browser may handle this differently.  Some experimentation may be required.

Furthermore, if MyLA is set up in an LMS course to open via LTI launch, but it is configured to open in a new window, it may appear to be more like a standalone application than an embedded one.  That is, the application opened in a new window may not be wrapped in an `iframe`.  Therefore, the fact that the application was launched via LTI would be **insufficient** to determine whether roles should be used.  It would be necessary to use one of these options:

* Determine whether the LTI launch specified embedded mode or new window.  There may be an LTI parameter to indicate this or it may be necessary to use the LMS' API.
* Use JavaScript or other browser technology to determine whether the application is being displayed as embedded.

## Recommendation

The ideal solution would be to determine whether roles in the MyLA page content embedded in the `iframe` interfere with roles specified by the LMS in parent elements.  If there is no interference, then MyLA's content should **_always_** specify roles.

Otherwise, if logic to detect embedded use is too complicated or delicate, add support for an LTI parameter to explicitly specify whether accessibility roles should be used.  (E.g., `useAccessibilityRoles=true`.)

## Resources
<details>
<summary>The 10/28/2019 accessibility review</summary>

Note that although the review was given on 10/28/2019, the files were mistakenly labelled as "10/28/18".

> * MyLA Accessibility Review 10/28/18
>   * Google original: https://docs.google.com/document/d/1IFBNWHvzY1l8YT7G1TX9jCrOyU9yAxF1-KazNbb4kc0
>   * GitHub PDF copy: https://github.com/tl-its-umich-edu/my-learning-analytics/files/5502938/MyLA.Accessibility.Review.10_28_18.pdf
> * MyLA Accessibility Table 10/28/18 - Issue Notes
>   * Google original: https://docs.google.com/spreadsheets/d/1ZWB6Ip4I9H8718G4uMn6VcRDrpS7dkS5BCR7nB0Q8ao
>   * GitHub PDF copy: https://github.com/tl-its-umich-edu/my-learning-analytics/files/5502940/MyLA.Accessibility.Table.10_28_18.-.Issue.Notes.pdf

</details>

**Comment by lsloan** [comment]
@gsilver and @ssciolla: Please read this issue and verify that I've captured the essence of the problem.  If it's not correct, please feel free to make corrections.
